### PR TITLE
Initial version

### PR DESCRIPTION
**Pull request by s-celles**

### CODE REVIEW

**Review by Tpuljak — CHANGES_REQUESTED** [review]
Screenshot 2024-11-28 at 17 15 56

@s-celles I get this error when running this in daytona (`daytona create https://github.com/daytonaio/sample-c/pull/1`).

You don't experience this issue?

File: .devcontainer/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,18 @@
+FROM mcr.microsoft.com/devcontainers/cpp:1-ubuntu-22.04
```
Comment (s-celles) [inline .devcontainer/Dockerfile]: 22.04 is pretty old.

according https://mcr.microsoft.com/en-us/artifact/mar/devcontainers/cpp/about

I should probably use 

    mcr.microsoft.com/devcontainers/cpp:ubuntu

Comment (mojafa) [reply .devcontainer/Dockerfile]: @s-celles did you make the changes? I'm still getting the same error.
Screenshot 2024-12-02 at 11 32 51

Comment (s-celles) [reply .devcontainer/Dockerfile]: I'm currently quite busy and can't handle this this week. Sorry

Comment (mojafa) [reply .devcontainer/Dockerfile]: @s-celles any updates?

Comment (s-celles) [reply .devcontainer/Dockerfile]: Should be ready

**Comment by mojafa** [comment]
@s-celles I couldn't run your project.

Kept on getting this error:
Screenshot 2024-12-04 at 12 50 46

I howevr did manage to make it run by:

1. removed all the comments in the devcontainer.json file
2. corrected the `postCreateCommand`. See below
Screenshot 2024-12-04 at 16 05 34

3. Changes image in dockerfile from `FROM mcr.microsoft.com/devcontainers/cpp:ubuntu` to `FROM ubuntu:22.04` and update accordingly on the devcontainer file. your image is bring incompatibility issues.
`Docker image mcr.microsoft.com/devcontainers/cpp:ubuntu is not available for linux/arm64/v8. 
`Screenshot 2024-12-04 at 13 55 08

4. ensure to update `.gitignore` file, once your code runs.  When it runs you should see thisbelow so add them to gitignore:
Screenshot 2024-12-04 at 14 02 31


5. also put `daytona` as the user in the dockerfile instead of `vscode`

**Comment by mojafa** [comment]
you can check my solution here: https://github.com/mojafa/sample-c

**Comment by s-celles** [comment]
22.04 is pretty old.
I'm fine with closing this PR and merge yours

**Comment by Tpuljak** [comment]
> I'm fine with closing this PR and merge yours

@s-celles we'll go with your solution since almost all of the code is yours and @mojafa simply made a fix for the devcontainer definition.

**Comment by s-celles** [comment]
ok so busy this week

**Comment by Tpuljak** [comment]
>  ok so busy this week

@s-celles no worries at all. Take your time.

**Comment by mojafa** [comment]
@s-celles any updates?

**Comment by s-celles** [comment]
Hi @mojafa 
this is working but I'm not a big fan of this image as it doesn't have tab completion in VS Code Terminal.
I also think we should have a clear definition of what features "ideal" Daytona sample projects should have.
unit tests / auto code formatting on save / linting ...

**Comment by mojafa** [comment]
@Tpuljak what do you think?

**Comment by Tpuljak** [comment]
@s-celles the tab completions work when you install the extensions listed in the devcontainer configuration.
Screenshot 2024-12-09 at 09 25 49

The issue is that we have a bug with that. We'll release a fix for that this week but we can proceed with this PR.
Screenshot 2024-12-09 at 09 28 56

File: .devcontainer/devcontainer.json [hunk 2]
```diff
@@ -0,0 +1,19 @@
+// For format details, see https://aka.ms/devcontainer.json. For config options, see the
```
Comment (Tpuljak) [inline .devcontainer/devcontainer.json:1]: Please format the file correctly. Indentation is not consistent.

File: .vscode/cmake-kits.json [hunk 3]
```diff
@@ -0,0 +1,18 @@
+[
```
Comment (Tpuljak) [inline .vscode/cmake-kits.json:1]: Please add this file into .vscode as well. It should be called `c_cpp_properties.json`.

```json
{
    "configurations": [
        {
            "name": "Linux",
            "includePath": [
                "${workspaceFolder}/**"
            ],
            "defines": [],
            "compilerPath": "/usr/bin/gcc",
            "cStandard": "c17",
            "cppStandard": "gnu++17",
            "intelliSenseMode": "linux-gcc-arm64",
            "configurationProvider": "ms-vscode.cmake-tools"
        }
    ],
    "version": 4
}
```

**Review by Tpuljak — CHANGES_REQUESTED** [review]

File: .vscode/cmake-kits.json (hunk 3 above)
Comment (s-celles) [reply .vscode/cmake-kits.json:1]: Should be done.
About tab completion I mean the completion in VS Code terminal
image

Comment (Tpuljak) [reply .vscode/cmake-kits.json:1]: Oh, that can be solved easily with the `common-utils` devcontainer feature. 

Example of how we set it up for Daytona: https://github.com/daytonaio/go-devcontainer/blob/main/.devcontainer/devcontainer.json#L10.

Comment (s-celles) [reply .vscode/cmake-kits.json:1]: Sorry but it won't be my top priority currently. Please consider merging as it is and add this in an another round that if you think that will be valuable.

Comment (Tpuljak) [reply .vscode/cmake-kits.json:1]: @s-celles no problem. That works.

@mojafa I'll merge this PR. Please raise a new one that will add the common utils to configure zsh.

**Pull request merged**